app/src/main/scala/org/alephium/explorer/util/UtxoUtil.scala

```diff
@@ -40,6 +40,44 @@ object UtxoUtil {
       .foldLeft(Option(U256.Zero)) { case (acc, amount) => acc.flatMap(_.add(amount)) }
   }
 
+  def tokenAmountForAddressInInputs(
+      address: Address,
+      inputs: ArraySeq[Input]
+  ): Map[TokenId, Option[U256]] = {
+    inputs
+      .filter(_.address == Some(address))
+      .flatMap(_.tokens)
+      .flatten
+      .groupBy(_.id)
+      .map { case (id, tokens) =>
+        (
+          id,
+          tokens.foldLeft(Option(U256.Zero)) { case (acc, token) =>
+            acc.flatMap(_.add(token.amount))
+          }
+        )
+      }
+  }
+
+  def tokenAmountForAddressInOutputs(
+      address: Address,
+      outputs: ArraySeq[Output]
+  ): Map[TokenId, Option[U256]] = {
+    outputs
+      .filter(_.address == address)
+      .flatMap(_.tokens)
+      .flatten
```

Comment:
[Nit] probably can have a function to add token amounts, which can be used for both inputs & outputs.